File: mobile/lib/models/rental.dart
class Rental {
  final int id;
  final int truckId;
  final int customerId;
  final String startDate;
  final String endDate;
  final double? pricePerDay;
  final double? totalPrice;
  final int? days;
  final String status;
  final String? notes;
  final String? createdAt;

  Rental({
    required this.id,
    required this.truckId,
    required this.customerId,
    required this.startDate,
    required this.endDate,
    this.pricePerDay,
    this.totalPrice,
    this.days,
    this.status = 'pending',
    this.notes,
    this.createdAt,
  });

  factory Rental.fromJson(Map<String, dynamic> json) {
    return Rental(
      id: json['id'],
      truckId: json['truck_id'],
      customerId: json['customer_id'],
      startDate: json['start_date'] ?? '',
      endDate: json['end_date'] ?? '',
      pricePerDay: json['price_per_day']?.toDouble(),
      totalPrice: json['total_price']?.toDouble(),
      days: json['days'],
      status: json['status'] ?? 'pending',
      notes: json['notes'],
      createdAt: json['created_at'],
    );
  }
}

File: backend/app/Models/Rental.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

#[Fillable([
    'truck_id',
    'customer_id',
    'start_date',
    'end_date',
    'price_per_day',
    'total_price',
    'status',
    'notes',
])]
class Rental extends Model
{
    /**
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'start_date' => 'date',
            'end_date' => 'date',
        ];
    }

    public function truck(): BelongsTo
    {
        return $this->belongsTo(Truck::class);
    }

    public function customer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'customer_id');
    }
}
